src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/CollectionControllerTest.php

```diff
@@ -1067,6 +1066,8 @@ public function testPutWithChildCollection()
             5
         );
 
+        $client = $this->createAuthenticatedClient();
```

Comment:
Why do we have to move it here? I think I've understood you differently the last time.

Reply:
`createAuthenticedClient` seems to destroy the old kernel and create a new one and createCollection uses the entityManager which was created before and seems to access the `kernel` service in some listener and as the kernel was destroyed before it end ups in: synthetic service `kernel` not found errror. To avoid this I create the collection before the client so the old kernel exist before. Basically its just a bad idea to bind services inside the setup method instead for symfony 4.3 it should `self::$container->get(..)` be used.

Reply:
So just that I get this right: We should now always access the container using the static variable, as described in [the Symfony documentation](https://symfony.com/doc/current/testing.html#accessing-the-container), right? The new Symfony documentation also gets the [client via a static method call](https://symfony.com/doc/current/testing.html#your-first-functional-test), how is that connected? Should we maybe also do that, and then it would not matter where the call took place?

Reply:
Yeah we can use the static variable and functions like its documented in the symfony documentation now. The difference between `getContainer` and `self::$container` is that `getContainer` would boot the kernel if not yet booted.